request/auth.php

```diff
@@ -14,7 +14,7 @@
                     case 'login':
                         if (!empty($_POST['email']) && !empty($_POST['password'])) {
                             if (FirstAide\Utils::isValidEmail($_POST['email'])) {
-                                $Auth = FirstAide\Authentication::withEmailPassword(
+                                $Auth = FirstAide\Authentication::createInstanceWithEmailPassword(
```

Comment:
I looked into this and found that only ajax call handlers should contain $_POST and $_GET. It's a good practice suggested in varies php frameworks such as laravel and Slim. There are 2 major reasons behind this. 
- 1) If functions/methods contain global variables like POST and GET, then using it in CLI is not possible. 
- 2) One such use case of CLI use is testing. Hence such methods and functions are loose their testability. It's suggested to treat functions and methods (member functions of a class) as a black box and only required and minimal variables must be passed to the functions.

Using global variables inside functions and methods give rise to an uncontrolled flow of data and give rise to non-standardized practices. Here are few links where other projects are doing same as we are doing. Please have look:

[1] https://github.com/owncloud/core/blob/f10d105ea2f2d0822693313ddcd6f9b2eaac6ebe/core/ajax/appconfig.php

[2] https://github.com/bcit-ci/CodeIgniter/blob/8186b6b398c26c0c1e0052eaddf4fc122e4929a5/system/helpers/form_helper.php (Codeigniter is also a framework)

[3]https://github.com/owncloud/core/blob/f10d105ea2f2d0822693313ddcd6f9b2eaac6ebe/apps/files_trashbin/ajax/delete.php

@npari @MehaKaushik @Anjali05